### PR TITLE
The new default `rich` server logger format emits unreadable logs

### LINKED ISSUE #1102: The new default "rich" server logger format emits unreadable logs
```
╱╱╭╮╱╱╭╮╱╱╱╱╱╱╭╮
╱╱┃┃╱╭╯╰╮╱╱╱╱╱┃┃
╭━╯┣━┻╮╭╋━━┳━━┫┃╭╮
┃╭╮┃━━┫┃┃╭╮┃╭━┫╰╯╯
┃╰╯┣━━┃╰┫╭╮┃╰━┫╭╮╮
╰━━┻━━┻━┻╯╰┻━━┻╯╰╯
╭━━┳━━┳━┳╮╭┳━━┳━╮
┃━━┫┃━┫╭┫╰╯┃┃━┫╭╯
┣━━┃┃━┫┃╰╮╭┫┃━┫┃
╰━━┻━━┻╯╱╰╯╰━━┻╯

✓ Applied ~/.dstack/server/config.yml

The admin token is be68ae95-050f-472f-84c1-5c25ac898b52
The dstack server (no version) is running at http://127.0.0.1:8000

[10:28:33] DEBUG    c dstack._internal.server.background.tasks.process_instance…
                    h                                                           
                    e                                                           
                    c                                                           
                    k                                                           
                                                                                
                    i                                                           
                    n                                                           
                    s                                                           
                    t                                                           
                    a                                                           
                    n                                                           
                    c                                                           
                    e                                                           
                                                                                
                    p                                                           
                    l                                                           
                    a                                                           
                    s                                                           
                    t                                                           
                    i                                                           
                    c                                                           
                    -                                                           
                    r                                                           
                    a                                                           
                    t                                                           
                    -                                                           
                    1                                                           
                    -                                                           
                    0                                                           
                    -                                                           
                    0                                                           
                                                                                
                    s                                                           
                    t                                                           
                    a                                                           
                    t                                                           
                    u                                                           
                    s                                                           
                    :                                                           
                                                                                
                    s                                                           
                    h                                                           
                    i                                                           
                    m                                                           
                                                                                
                    h                                                           
                    e                                                           
                    a                                                           
                    l                                                           
                    t                                                           
                    h                                                           
                                                                                
                    i                                                           
                    s                                                           
                                                                                
                    O                                                           
                    K       
```

### PR DESCRIPTION
**Pull request by peterschmidt85**
Fixes #1102

### CODE REVIEW

File: src/dstack/_internal/server/app.py [hunk 1]
```diff
@@ -94,16 +94,13 @@ async def lifespan(app: FastAPI):
...
                     await server_config_manager.init_config(session=session)
                 console.print(
-                    f"[code]✓[/] Initialized the default configuration at [code]{server_config_dir}[/]"
+                    f"[code]✓[/] Initialized the default configuration at [code]{server_config_dir}[/]\n"
                 )
             else:
                 with console.status(f"Applying [code]{server_config_dir}[/]..."):
                     await server_config_manager.apply_config(session=session, owner=admin)
-                console.print(f"[code]✓[/] Applied [code]{server_config_dir}[/]")
-        with console.status("Initializing gateways..."):
```
Comment (jvstme) [inline src/dstack/_internal/server/app.py:103]: Why remove the throbber here? Gateway init can take a few seconds (7 for me), so it seems reasonable to have a throbber

File: src/dstack/_internal/server/app.py
Comment (jvstme) [inline src/dstack/_internal/server/app.py:1]: I'd suggest moving unrelated changes to separate PRs, even if they are pretty small. They still make the review slower and can block the PR if there are any issues with them

File: src/dstack/_internal/cli/utils/rich.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+            time_format=time_format,
+            level=level,
+            path=path,
+            line_no=record.lineno,
+            link_path=record.pathname if self.enable_link_path else None,
+        )
+        return log_renderable
+
+    def _lazy_init_render(self):
+        if self.__log_render is None:
```
Comment (jvstme) [inline src/dstack/_internal/cli/utils/rich.py]: Is this necessary? Initializing `__log_render` on class level should work too.
```python
class DstackRichHandler(RichHandler):
    __log_render = LogRender()
```

File: src/dstack/_internal/cli/utils/rich.py [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    from rich.console import Console, ConsoleRenderable, RenderableType
+    from rich.table import Table
+
+FormatTimeCallable = Callable[[datetime], Text]
+
+
+class LogRender:
+    def __init__(
+        self,
+        show_time: bool = True,
```
Comment (jvstme) [inline src/dstack/_internal/cli/utils/rich.py]: (nit) I understand this is copied from rich, but since we only initialize `LogRender()` without arguments, we could remove all the parameters and hence simplify the implementation of `__call__`

File: src/dstack/_internal/cli/utils/rich.py [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+        from rich.containers import Renderables
+        from rich.table import Table
+
+        output = Table.grid(padding=(0, 1))
+        output.expand = True
+        if self.show_time:
+            output.add_column(style="log.time")
+        if self.show_level:
+            output.add_column(style="log.level", width=self.level_width)
+        output.add_column(ratio=1, overflow="ignore")
```
Comment (jvstme) [inline src/dstack/_internal/cli/utils/rich.py]: Can't see the logs in a narrow terminal window, messages are cut. Maybe this line is the reason
trimmed-logs

**Review by jvstme — CHANGES_REQUESTED** [review]
Logs look nice, thank you! I found a couple of issues though, please see my comments

**Comment by peterschmidt85** [comment]
@jvstme Please review and merge if possible!

**Review by jvstme — COMMENTED** [review]
Message grouping by time does not look right.
grouping-current
Compare to RichHandler.
grouping-default
